src/chapters/control-flow/loops.rst

```diff
@@ -0,0 +1,278 @@
+Loops
+=====
+
+.. index:: ! for loop
+
+``for`` Loop
+------------
+
+In Java we write a definite loop (aka a **for loop**) as:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (int i = 0; i < 10; i++ ) {
+      System.out.println(i);
+   }
+
+.. note:: 
+
+   You may not be familiar with the expression ``i++`` since it is not
+   found in all languages. The ``++`` is an increment operator that has the same
+   effect as ``i += 1``. In this example, since the ``++`` comes after
+   ``i``, we call it a postfix increment operator. There is also a ``--``
+   decrement operator in Java. For more information, see 
+   `Increment and Decrement Operators <http://www.javawithus.com/tutorial/increment-and-decrement-operators>`__.
+
+
+The Java ``for`` loop gives you explicit control over the starting, stopping, and stepping 
+of the loop variable inside the parentheses. You can think of it this way:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (start clause; stop clause; step clause) {
+      statement1
+      statement2
+      ...
+   }
+
+If you want to start at 100, stop at 0 and count backward by 5, the loop is written as:
+
+.. sourcecode:: java
+   :linenos:
+
+   for (int i = 100; i >= 0; i -= 5) {
+      System.out.println(i);
+   }
+
+.. index:: ! for-each loop
+
+``for-each`` Loop
+-------------
+
+Java also provides a syntax to iterate over any sequence or collection, such as an Array: 
+
+.. sourcecode:: java
+   :linenos:
+
+   int k[] = {1, 1, 2, 3, 5, 8, 13, 21};
+
+   for (int i : k) {
+      System.out.println(i);
+   }
+
+Here, the loop variable moves through the items in the Array of integers, ``k[]``. The syntax
+here uses a colon symbol, ``:``. This type of loop is known as a **for-each loop**.
+
+.. tip::
+
+   When considering this structure, it can be helpful to read the code sample above to yourself
+   as "For each integer in ``Array k``...".
+
+This loop version also works with a String, where we
+can convert the String to an Array of characters:
+
+.. sourcecode:: java
+   :linenos:
+
+   String msg = "Hello World";
+
+   for (char c : msg.toCharArray()) {
+      System.out.println(c);
+   }
+
+As you see, to iterate through a String in this way, Java requires an extra String method,
+``.toCharArray()``, to convert the String to an Array of characters.
+
+.. index:: ! while loop
+
+``while`` Loop
+--------------
+
+Java also supports the **while loop**, or indefinite loop.
+Recall that in Python the ``while`` loop is written as:
```

Comment:
Remove the Python reference.  Maybe replace it with a reminder of the difference between a *definite* loop and an *indefinite* one.